src/types/num.rs

```diff
@@ -10,3 +13,299 @@ use crate::{ffi, PyAny};
 pub struct PyLong(PyAny);
 
 pyobject_native_type_core!(PyLong, pyobject_native_static_type_object!(ffi::PyLong_Type), #checkfunction=ffi::PyLong_Check);
+
+impl PyLong {
+    /// Creates a new `PyLong` from a C `long` integer.
+    /// # Returns
+    /// A [`Bound`] reference to a `PyLong` object representing the given `val`.
+    #[inline]
+    pub fn new_bound_from_c_long(py: Python<'_>, val: c_long) -> Bound<'_, Self> {
+        unsafe {
+            ffi::PyLong_FromLong(val)
+                .assume_owned(py)
+                .downcast_into_unchecked()
+        }
+    }
+}
```

Comment:
I don't think we should have this, at least not in this form. See https://github.com/PyO3/pyo3/issues/2221 for the prior discussion on this function.

Reply:
Thank you for the feedback! I'm still learning Rust, I greatly appreciate any help!

Before implementing `PartialEq`, I believe it's necessary to define a the constructor. In this regard, I have two potential approaches in mind: using the `num` crate to create a generic `new_bound` function, or defining an internal trait specific to this file.

What would you recommend as the best approach for this situation?

Reply:
> Before implementing PartialEq, I believe it's necessary to define a the constructor.

How come? You dont need to create them to compare them. Anyway, the current way to create a PyLong is through `IntoPy`, so (strictly speaking) this function is not needed.

Reply:
Thanks! Removed the constructor!